controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -92,7 +99,360 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+func (r *Reconciler) updateExistingSpaceBinding(logger logr.Logger, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest, spaceBinding *toolchainv1alpha1.SpaceBinding) (bool, error) {
+	// check if spacebinding is being deleted
+	if util.IsBeingDeleted(spaceBinding) {
+		return false, fmt.Errorf("cannot update SpaceBinding because it is currently being deleted")
+	}
+	return r.updateSpaceBinding(logger, spaceBinding, spaceBindingRequest)
+}
+
+// updateSpaceBinding updates the MUR and Role from the spaceBindingRequest to the spacebinding object
+// if they are not up-to-date.
+// returns false/nil if everything is up-to-date
+// returns true/nil if spacebinding was updated
+// returns false/err if something went wrong
+func (r *Reconciler) updateSpaceBinding(logger logr.Logger, spaceBinding *toolchainv1alpha1.SpaceBinding, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) (bool, error) {
+	logger.Info("update spaceBinding")
+	if spaceBindingRequest.Spec.MasterUserRecord == spaceBinding.Spec.MasterUserRecord &&
+		spaceBindingRequest.Spec.SpaceRole == spaceBinding.Spec.SpaceRole {
+		// everything is up-to-date let's return
+		return false, nil
+	}
+
+	// update SpaceRole and MUR
+	logger.Info("updating spaceBinding", "spaceBinding.Name", spaceBinding.Name)
+	spaceBinding.Spec.SpaceRole = spaceBindingRequest.Spec.SpaceRole
+	spaceBinding.Spec.MasterUserRecord = spaceBindingRequest.Spec.MasterUserRecord
+	// update also the MUR label
+	spaceBinding.Labels[toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey] = spaceBindingRequest.Spec.MasterUserRecord
+	err := r.Client.Update(context.TODO(), spaceBinding)
+	if err != nil {
+		return false, errs.Wrap(err, "unable to update SpaceRole and MasterUserRecord fields")
+	}
+
+	logger.Info("spaceBinding updated", "spaceBinding.name", spaceBinding.Name, "spaceBinding.Spec.Space", spaceBinding.Spec.Space, "spaceBinding.Spec.SpaceRole", spaceBinding.Spec.SpaceRole, "spaceBinding.Spec.MasterUserRecord", spaceBinding.Spec.MasterUserRecord)
+	return true, nil
+}
+
+func (r *Reconciler) createNewSpaceBinding(logger logr.Logger, memberCluster cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest, mur *toolchainv1alpha1.MasterUserRecord, space *toolchainv1alpha1.Space) error {
+	spaceBinding := spacebinding.NewSpaceBinding(mur, space, spaceBindingRequest.Name)
+	// set SBR labels on spacebinding
+	spaceBinding.Labels[toolchainv1alpha1.SpaceBindingRequestLabelKey] = spaceBindingRequest.GetName()
+	spaceBinding.Labels[toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey] = spaceBindingRequest.GetNamespace()
+	logger.Info("creating spacebinding", "spaceBinding.Name", spaceBinding.Name)
+	if err := r.setStatusProvisioning(memberCluster, spaceBindingRequest); err != nil {
+		return err
+	}
+	if err := r.Client.Create(context.TODO(), spaceBinding); err != nil {
+		return errs.Wrap(err, "unable to create SpaceBinding")
+	}
```

Comment:
btw, we should take care of conflicts with the already existing (default) SpaceBindings. Or with those that are created by admins. Such SpaceBindings are immutable for end users, so it would be good to set corresponding status in SBR CR.
It's fine if you do it in a new PR.

Reply:
If I'm not mistaken this is already coverd by the fact that the name of the SpaceBinding is computed here https://github.com/codeready-toolchain/toolchain-common/blob/c909c72d8f627a9f72917531eded2a5e765faea1/pkg/spacebinding/spacebinding.go#L24

So in case an SBR tries to create a SB resource for mur and space that already has a SpaceBinding, the Create will fail saying that the name already exists and it will be set in the SBR status : https://github.com/codeready-toolchain/host-operator/pull/848/files#diff-e48d49741db80226348ae5ffd08100dd10afa2e83e5d18b2a94776dfdfb616f3R250-R252

Reply:
Ok, then let's do it in separate PR, because if we want to specify if the SB that already exists is or not managed by SBR controller , we need to get it and check the labels. Not sure if we want to provide such information to the user/client?